### ISSUE TITLE
excludeAllQueryParams when adding a query param

### ISSUE THREAD

**Issue by deanmarano** [ember-a11y/ember-a11y-refocus#378]
Thanks for the recent addition of the `excludeAllQueryParams` parameter! This fits most of our needs, but we're missing a test case.

When loading a page, we start with no query params. When a user start searching in a search input, the `search` query param is added and populated. If the user the clears the search, focus is changed from the search input to the nav.

An example test case might look like:

```
    test('excludeAllQueryParams is true, queryParams do not exist but are added, do not manage focus', async function (assert) {
      let router = this.owner.lookup('service:router');

      await render(hbs`<NavigationNarrator @excludeAllQueryParams={{true}} />`);
      router.trigger('routeDidChange', new MockTransition({
          from: new MockRouteInfo({
            name: 'biscuit',
            params: { id: 'hobnob' },
          }),
          to: new MockRouteInfo({
            name: 'biscuit',
            params: { id: 'hobnob' },
            queryParams: { region: 'apac' },
          }),
        })

      await settled();

      assert.dom('#ember-a11y-refocus-nav-message').isNotFocused();
    });
```

It seems that [this line](https://github.com/ember-a11y/ember-a11y-refocus/pull/367/files#diff-61bb4f399b0127fee433f89d5d6e5be3a3f02ff2cdd76e5694a07e748c3f80dcR111-R112) is causing the issue - it would be great if a change from no params to params would follow the `excludeAllQueryParams` param.

**Comment by MelSumner** [comment]
Thank you for reporting this issue, apologies for the inconvenience. Will fix ASAP!

**Comment by MelSumner** [comment]
Just to document while I figure it out: 

1. I can reproduce this with a failing test
2. Looking at the code,  https://github.com/ember-a11y/ember-a11y-refocus/blob/7d20ca2f5d82b24d13d550257d4a61d0e64de7f9/addon/components/navigation-narrator.js#L82 should catch the scenario 🤔  Or rather, it was intended to catch the scenario

Will need to ask around a little and dig into it some more, but ideas welcome.

**Comment by MelSumner** [comment]
I merged the branch to main but I don't want to cut a new release until we're really sure this won't mess up other scenarios.

**Comment by MelSumner** [comment]
Resolved in v4.1.3
